src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -170,16 +246,16 @@ end
     niter = 1
     ninners = zeros(Int,0)
     while true
-        BYX = BY'X
-        # XXX the one(T) instead of plain old 1 is because of https://github.com/JuliaArrays/BlockArrays.jl/issues/176
-        mul!(X, Y, BYX, -one(T), one(T)) # X -= Y*BY'X
+        BYX = block_overlap(BY,X) # = BY' X
+        X .-= block_mul(Y, BYX) #X = X -Y * BYX
```

Comment:
this is a bit less efficient than the previous version. Maybe have the custom type override `mul!(a,b,c,alpha,beta)`?